Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/person/ui/details/PersonDetailsViewModel.kt

```diff
@@ -102,7 +102,10 @@ class PersonDetailsViewModel @AssistedInject constructor(
             // Find any vaccination certificate to determine the vaccination information
             personCertificates.certificates.find { it is VaccinationCertificate }?.let { certificate ->
                 val vaccinatedPerson = vaccinatedPerson(certificate)
-                if (vaccinatedPerson != null && vaccinatedPerson.getVaccinationStatus(timeStamper.nowUTC) != IMMUNITY) {
+                if (vaccinatedPerson != null &&
+                    vaccinatedPerson.getVaccinationStatus(timeStamper.nowUTC) != IMMUNITY &&
+                    !vaccinatedPerson.isFirstVaccinationDoseAfterRecovery()
```

Comment:
Please write some unit tests?

Reply:
I actually rather change the logic of `VaccinatedPerson.getVaccinationStatus()` to get the status without having to call `isFirstVaccinationDoseAfterRecovery()` here , so maybe we just need to change line   **(you need to check what is best here )**
https://github.com/corona-warn-app/cwa-app-android/blob/316bd648c326d5e97408d749952778165235bd48/Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/vaccination/core/VaccinatedPerson.kt#L57

To

```kotlin  
daysToImmunity <= 0 || isFirstVaccinationDoseAfterRecovery() -> Status.IMMUNITY
 ```
by doing that :
- Persons who got these kind of vaccine are Immune , right ? and  Immune state is used in details screen to display different colouring. 
- the state will be shared and changed in one place 

What are you thoughts on this?